theano/gof/cc.py

```diff
@@ -547,9 +547,7 @@ def accept(self, fgraph, no_recycling=None):
         if no_recycling is None:
             no_recycling = []
         if self.fgraph is not None and self.fgraph is not fgraph:
-            return type(self)().accept(fgraph, no_recycling)
-            # raise Exception("Cannot accept from a Linker that is already"
-            #                " tied to another FunctionGraph.")
+            return type(self)(self.schedule).accept(fgraph, no_recycling)
```

Comment:
Maybe add a comment explaining that we have to create a new `Linker` instance, because a Linker can only be tied to one FunctionGraph.